rust/gui-client/src-tauri/src/client/gui.rs

```diff
@@ -740,14 +742,29 @@ async fn run_controller(
         network_changes::Worker::new().context("Failed to listen for network changes")?;
 
     let mut dns_listener = network_changes::DnsListener::new()?;
+    let mut last_resolvers_sent = vec![];
 
     loop {
         tokio::select! {
             () = controller.notify_controller.notified() => {
+                let tunnel_became_ready = !controller.tunnel_ready;
+                controller.tunnel_ready = true;
                 tracing::debug!("Controller notified of new resources");
                 if let Err(error) = controller.refresh_system_tray_menu() {
                     tracing::error!(?error, "Failed to reload resource list");
                 }
+                if tunnel_became_ready {
```

Comment:
I can't do the first part because refreshing the tray menu needs `tunnel_ready` to be up to date. I'll do the second part and leave a comment explaining the reasoning for the first part